zjit/src/hir.rs

```diff
@@ -1417,6 +1427,7 @@ impl Function {
             Insn::HashDup { .. } => types::HashExact,
             Insn::NewRange { .. } => types::RangeExact,
             Insn::NewRangeFixnum { .. } => types::RangeExact,
+            Insn::ObjectAlloc { .. } => types::HeapObject,
```

Comment:
Let's specialize this with the class of passed into `ObjectAlloc`! Then we could get rid of a `GuardType` here

Reply:
sounds good :+1: